src/QMCWaveFunctions/EinsplineSetBuilderESHDF.fft.cpp

```diff
@@ -73,20 +71,14 @@ bool EinsplineSetBuilder::ReadOrbitalInfo_ESHDF(bool skipChecks)
   int have_dpsi         = false;
   int NumAtomicOrbitals = 0;
   NumCoreStates = NumMuffinTins = NumTwists = NumSpins = NumBands = NumAtomicOrbitals = 0;
-  //vector<int> nels_spin(2);
-  //nels_spin[0]=TargetPtcl.last(0)-TargetPtcl.first(0);
-  //nels_spin[1]=TargetPtcl.getTotalNum()-nels_spin[0];
   NumElectrons = TargetPtcl.getTotalNum();
-  HDFAttribIO<int> h_NumBands(NumBands), h_NumSpins(NumSpins), h_NumTwists(NumTwists), h_NumCore(NumCoreStates),
-      h_NumMuffinTins(NumMuffinTins), h_have_dpsi(have_dpsi), h_NumAtomicOrbitals(NumAtomicOrbitals);
-  h_NumBands.read(H5FileID, "/electrons/kpoint_0/spin_0/number_of_states");
-  h_NumCore.read(H5FileID, "/electrons/kpoint_0/spin_0/number_of_core_states");
-  //h_NumElectrons.read  (H5FileID, "/electrons/number_of_electrons");
-  h_NumSpins.read(H5FileID, "/electrons/number_of_spins");
-  h_NumTwists.read(H5FileID, "/electrons/number_of_kpoints");
-  h_NumMuffinTins.read(H5FileID, "/muffin_tins/number_of_tins");
-  h_have_dpsi.read(H5FileID, "/electrons/have_dpsi");
-  h_NumAtomicOrbitals.read(H5FileID, "/electrons/number_of_atomic_orbitals");
+  H5File.read(NumBands, "/electrons/kpoint_0/spin_0/number_of_states");
+  H5File.readEntry(NumCoreStates, "/electrons/kpoint_0/spin_0/number_of_core_states");
```

Comment:
How did read or readEntry get chosen?
read is used when the read must succeed.
readEntry leaves the check to user code. So some optional data reading needs it.
Did  you first try all with read() and then gradually replace read with readEntry when optional data doesn't exist in tests?

Reply:
Yes, I initially tried everything with hdf_archive::read and changed calls to hdf_archive::readEntry until all of deterministic tests pass.